truffleruby/src/main/ruby/core/truffle/interop.rb

```diff
@@ -63,7 +63,29 @@ def self.enumerable(foreign)
     end
 
     class Foreign
-
+      JAVA_CLASS_CLASS = Foreign.java_class_by_name("java.lang.Class")
+      JAVA_OBJECT_CLASS = java_class_by_name("java.lang.Object")
+      JAVA_FIELD_CLASS = Foreign.java_class_by_name("java.lang.reflect.Field")
+      JAVA_STRING_CLASS = Foreign.java_class_by_name("java.lang.String")
+      
+      CLASS_GET_FIELD = Foreign.get_java_method(
+        JAVA_CLASS_CLASS, "getField", false, JAVA_FIELD_CLASS, JAVA_STRING_CLASS)
+      FIELD_GET = Foreign.get_java_method(
+        JAVA_FIELD_CLASS, "get", false, JAVA_OBJECT_CLASS, JAVA_OBJECT_CLASS)
+      
+      JAVA_INTEGER_CLASS = Foreign.java_class_by_name("java.lang.Integer")
+      JAVA_PRIM_INT_CLASS = Foreign.invoke_java_method(
+        FIELD_GET,
+        Foreign.invoke_java_method(CLASS_GET_FIELD,
+                                   JAVA_INTEGER_CLASS,
+                                   Foreign.to_java_string("TYPE")),
+        nil)
+      OBJECT_HASH_CODE = get_java_method(
+        JAVA_OBJECT_CLASS, "hashCode", false, JAVA_PRIM_INT_CLASS)
```

Comment:
@chrisseaton all the bootstrap of this in Jruby is in Java with their own annotations for creating Jruby classes and methods. If we use that then we're going to be suffering from the same large amount of Java code they have, and it's quite deeply tied to their use of Java reflection to implement everything.